### PR TITLE
Adjust `isFusableUsingTileAndFuse` in `SinkReshapes`

### LINKED ISSUE #18893: [DispatchCreation] `tensor.expand_shape` blocks `softmax` -> "bit truncate" fusion
In the example below, `%2` and `%3` are placed into separate dispatches:

```mlir
util.func public @main(%arg0: tensor<4x64x?xf16>, %arg1: tensor<4xi64>) -> tensor<4x64x1x?xf16> {
  %cst = arith.constant 0xFC00 : f16
  %cst_0 = arith.constant 0.000000e+00 : f16
  %cst_1 = arith.constant 11.3137083 : f32
  %c2 = arith.constant 2 : index
  %dim = tensor.dim %arg0, %c2 : tensor<4x64x?xf16>
  %0 = tensor.empty(%dim) : tensor<4x64x?xf32>
  %1 = linalg.generic {indexing_maps = [affine_map<(d0, d1, d2) -> (d0, d1, d2)>, affine_map<(d0, d1, d2) -> (d0)>, affine_map<(d0, d1, d2) -> (d0, d1, d2)>], iterator_types = ["parallel", "parallel", "parallel"]} ins(%arg0, %arg1 : tensor<4x64x?xf16>, tensor<4xi64>) outs(%0 : tensor<4x64x?xf32>) {
  ^bb0(%in: f16, %in_2: i64, %out: f32):
    %5 = linalg.index 2 : index
    %6 = arith.index_cast %5 : index to i64
    %7 = arith.cmpi sge, %6, %in_2 : i64
    %8 = arith.select %7, %cst, %cst_0 : f16
    %9 = arith.truncf %cst_1 : f32 to f16
    %10 = arith.divf %in, %9 : f16
    %11 = arith.addf %10, %8 : f16
    %12 = arith.extf %11 : f16 to f32
    linalg.yield %12 : f32
  } -> tensor<4x64x?xf32>
  %2 = linalg.softmax dimension(2) ins(%1 : tensor<4x64x?xf32>) outs(%0 : tensor<4x64x?xf32>) -> tensor<4x64x?xf32>
  %expanded = tensor.expand_shape %2 [[0], [1, 2], [3]] output_shape [4, 64, 1, %dim] : tensor<4x64x?xf32> into tensor<4x64x1x?xf32>
  %3 = tensor.empty(%dim) : tensor<4x64x1x?xf16>
  %4 = linalg.generic {indexing_maps = [affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>, affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%expanded : tensor<4x64x1x?xf32>) outs(%3 : tensor<4x64x1x?xf16>) {
  ^bb0(%in: f32, %out: f16):
    %5 = arith.truncf %in : f32 to f16
    linalg.yield %5 : f16
  } -> tensor<4x64x1x?xf16>
  util.return %4 : tensor<4x64x1x?xf16>
}
```

 `iree-compile in.mlir --compile-to=dispatch-creation`

### PR DESCRIPTION
**Pull request by IanWood1**
Adjust `isFusableUsingTileAndFuse` to return true if the producer implements `LinalgExt::LinalgFusionOpInterface`. This is motivated by a `tensor.expand_shape` getting stuck between `linalg.softmax` and a 'bit-truncate` op, preventing fusion and leading to materialization of the higher bit width tensor.

Side note: I'm not sure why these aren't being generalized? If they were we could bubble the `tensor.expand_shape` through the softmax

Closes https://github.com/iree-org/iree/issues/18893

### CODE REVIEW

**Review by MaheshRavishankar — APPROVED** [review]
Add a test?

And yes, we should just generalize softmax. I think that was a bit of legacy. Only issue is I think fusion of softmax automatically is gated by the `aggressiveFusion` flag.

**Comment by IanWood1** [comment]
Merged with main because I just pushed to main & adjusted dispatch count

File: .github/workflows/pkgci_regression_test.yml [hunk 1]
```diff
@@ -222,7 +222,7 @@ jobs:
             --goldentime-rocm-vae-ms 337.0 \
             --goldendispatch-rocm-unet 1531 \
             --goldendispatch-rocm-clip 1139 \
-            --goldendispatch-rocm-vae 246 \
+            --goldendispatch-rocm-vae 245 \
```
Comment (MaheshRavishankar) [inline .github/workflows/pkgci_regression_test.yml:225]: woohoo! saved a dispatch :D

Comment (IanWood1) [reply .github/workflows/pkgci_regression_test.yml:225]: yay! :tada:

**Pull request merged**